editor/selectors.js

```diff
@@ -368,8 +377,47 @@ export function getEditedPostPreviewLink( state ) {
  * @param  {String} uid   Block unique ID
  * @return {Object}       Parsed block object
  */
-export function getBlock( state, uid ) {
-	return state.editor.blocksByUid[ uid ];
+export const getBlock = createSelector(
+	( state, uid ) => {
+		const block = state.editor.blocksByUid[ uid ];
+		const type = getBlockType( block.name );
+
+		if ( ! block || ! type || ! type.attributes ) {
+			return block;
+		}
+
+		const metaAttributes = reduce( type.attributes, ( result, value, key ) => {
+			if ( value && 'meta' in value ) {
+				result[ key ] = getPostMeta( state, value.meta );
+			}
+
+			return result;
+		}, {} );
+
+		// Avoid injecting an empty `attributes: {}`
+		if ( ! block.attributes && ! metaAttributes.length ) {
```

Comment:
I'm confused how this is working, since `metaAttributes.length` will always be `undefined`, and this will shortcut before merging in attributes. I assume this should have been `Object.keys( metaAttributes ).length`.

Reply:
Also, shouldn't we still want to merge meta attributes even if there are no other attributes of the block?